src/main/java/com/takipi/oss/storage/resources/fs/multifetcher/MultiFetcher.java

```diff
@@ -14,12 +14,27 @@
 import java.util.List;
 
 public class MultiFetcher {
-    
+
+    private static class PartSizeEstimator {
+        private long totalSizeLoaded = 0;
+        private long numberOfPartsLoaded = 0;
+        synchronized void updateStats(long size) {
+            if (numberOfPartsLoaded < 1000000) {
```

Comment:
Why is this limit?

Reply:
To prevent integer overflow. I should have actually put the limit on totalSizeLoaded instead.